api/server/router/image/image_routes.go

```diff
@@ -226,16 +226,21 @@ func (s *imageRouter) getImagesJSON(ctx context.Context, w http.ResponseWriter,
 	if err != nil {
 		return err
 	}
-
-	version := httputils.VersionFromContext(ctx)
-	if versions.LessThan(version, "1.41") {
+	var sharedSize bool
+	if versions.LessThan(httputils.VersionFromContext(ctx), "1.41") {
 		filterParam := r.Form.Get("filter")
 		if filterParam != "" {
 			imageFilters.Add("reference", filterParam)
 		}
+	} else {
+		sharedSize = httputils.BoolValue(r, "shared-size")
```

Comment:
It's somewhat "obvious", but perhaps good to add a comment here, a-la;

```suggestion
		// support for the "shared-size" parameter was added in API 1.42
		sharedSize = httputils.BoolValue(r, "shared-size")
```